FirebaseAuth/Sources/Swift/Auth/Auth.swift

```diff
@@ -169,20 +169,29 @@ extension Auth: AuthInterop {
   /** @property app
    @brief Gets the `FirebaseApp` object that this auth object is connected to.
    */
-  @objc public weak var app: FirebaseApp?
+  @objc public internal(set) weak var app: FirebaseApp?
 
   /** @property currentUser
    @brief Synchronously gets the cached current user, or null if there is none.
    */
-  @objc public var currentUser: User?
+  @objc public internal(set) var currentUser: User?
 
   /** @property languageCode
    @brief The current user language code. This property can be set to the app's current language by
    calling `useAppLanguage()`.
 
    @remarks The string used to set this property must be a language code that follows BCP 47.
    */
-  @objc public var languageCode: String?
+  @objc public var languageCode: String? {
+    get {
+      return requestConfiguration.languageCode
```

Comment:
This needs to be lock-protected if the `set` is nonatomic.

Reply:
Good catch! This is a bug in the Objective C implementation.